src/libstore/globals.hh

```diff
@@ -116,6 +116,13 @@ public:
         "The number of lines of the tail of "
         "the log to show if a build fails."};
 
+    Setting<std::string> logFormat{this, "bar", "log-format",
+        R"(
+          The default log format to display when buuilding; options are `raw`,
+          `raw-with-logs`, `internal-json`, `bar`, and `bar-with-logs`.  'Bar'
+          refers to the progress bar showing the build progress."
```

Comment:
Maybe it's better to just document every logger instead of just `bar` in a list.

Reply:
> Shouldn't the `--log-format` option in src/libmain/common-args.cc be removed?

Sure, I will update this PR to remove it.

> Note that _not_ having this as an option was intentional. It seems dangerous (e.g. for scripts) to have config settings that globally change the output format of a command.

The output of `nix build` is on stderr.  I think it's sort of a convention that the stderr from any program is for humans, and you should not assume the ability to parse it.

[`NO_COLOR`](https://no-color.org/) is widely implemented and has this property.

Reply:
Another possibility -- which I don't recommend -- would be to ignore `log-format` if [`isatty(3)`](https://man7.org/linux/man-pages/man3/isatty.3.html) reports that stderr is _not_ a tty.  This would give deterministic behavior for scripts running unattended.

Personally I'm not a big fan of programs that change their behavior just because I piped their output through `grep`.